lib/python/qmk/cli/via2json.py

```diff
@@ -0,0 +1,163 @@
+        for key in layer:
+            if sorting_hat[pos][0] is not None:
+                layer_data.append([sorting_hat[pos][0], key])
+            pos += 1
+        layer_data.sort()
+        layer_data = [kc[1] for kc in layer_data]
+        keymap_data.append(layer_data)
+
+    return keymap_data
+
+
+@cli.argument('-o', '--output', arg_only=True, type=qmk.path.normpath, help='File to write to')
+@cli.argument('-q', '--quiet', arg_only=True, action='store_true', help="Quiet mode, only output error messages")
+@cli.argument('filename', type=qmk.path.FileType('r'), arg_only=True, help='VIA Backup JSON file')
+@cli.argument('-kb', '--keyboard', arg_only=True, required=True, help='The keyboard\'s name')
+@cli.argument('-km', '--keymap', arg_only=True, default='via2json', help='The keymap\'s name')
+@cli.argument('-l', '--layout', arg_only=True, help='The keymap\'s layout')
+@cli.subcommand('Convert a VIA backup json to keymap.json format.')
+def via2json(cli):
+    """Convert a VIA backup json to keymap.json format.
+
+    This command uses the `qmk.keymap` module to generate a keymap.json from a VIA backup json. The generated keymap is written to stdout, or to a file if -o is provided.
+    """
+    # Error checking
+    if not qmk.path.is_keyboard(cli.args.keyboard):
+        cli.log.error(f'Keyboard {cli.args.keyboard} does not exist!')
+        exit(1)
+
+    # Find appropriate layout macro
+    keymap_layout = cli.args.layout if cli.args.layout else _find_via_layout_macro(cli.args.keyboard)
+    if not keymap_layout:
+        cli.log.error(f"Couldn't find LAYOUT macro for keyboard {cli.args.keyboard}. Please specify it with the '-l' argument.")
+        exit(1)
+
+    # Environment processing
+    if cli.args.output and cli.args.output.name == '-':
+        cli.args.output = None
+
+    # Load the VIA backup json
+    with cli.args.filename.open('r') as fd:
+        via_backup = json.load(fd)
+
+    # Generate keyboard metadata
+    keyboard_data = info_json(cli.args.keyboard)
+
+    # Get keycode array
+    keymap_data = _via_to_keymap(via_backup, keyboard_data, keymap_layout)
+
+    # Convert macros
+    macro_data = list()
+    if via_backup.get('macros'):
+        macro_data = _convert_macros(via_backup['macros'])
+
+        # Replace VIA macro keys with JSON keymap ones
+        keymap_data = _fix_macro_keys(keymap_data)
+
+    # Generate the keymap.json
+    keymap_json = generate_json(cli.args.keymap, cli.args.keyboard, keymap_layout, keymap_data, macro_data)
+
+    if cli.args.output:
+        cli.args.output.parent.mkdir(parents=True, exist_ok=True)
+        if cli.args.output.exists():
+            cli.args.output.replace(cli.args.output.parent / (cli.args.output.name + '.bak'))
+        cli.args.output.write_text(json.dumps(keymap_json, cls=KeymapJSONEncoder))
+
+        if not cli.args.quiet:
+            cli.log.info('Wrote keymap to %s.', cli.args.output)
+
+    else:
+        print(json.dumps(keymap_json))
```

Comment:
With <https://github.com/qmk/qmk_firmware/pull/16674>, maybe something like
```suggestion
    lines = [json.dumps(keymap_json, cls=KeymapJSONEncoder)]
    dump_lines(cli.args.output, keymap_lines)
```

Reply:
`dump_lines` seems to be working nicely.